pkg/codegen/go/gen_crd2pulumi.go

```diff
@@ -0,0 +1,69 @@
+package gen
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"path"
+
+	"github.com/pulumi/pulumi/pkg/v2/codegen/schema"
+)
+
+// GenCRDTypes returns a map from each module name to a buffer containing the
+// code for its generated types.
+func GenCRDTypes(tool string, pkg *schema.Package) (map[string]*bytes.Buffer, error) {
+	if err := pkg.ImportLanguages(map[string]schema.Language{"go": Importer}); err != nil {
+		return map[string]*bytes.Buffer{}, err
+	}
+
+	goInfo, _ := pkg.Language["go"].(GoPackageInfo)
+	packages := generatePackageContextMap(tool, pkg, goInfo)
+
+	var pkgMods []string
+	for mod := range packages {
+		pkgMods = append(pkgMods, mod)
+	}
+
+	buffers := map[string]*bytes.Buffer{}
+
+	for _, mod := range pkgMods {
+		pkg := packages[mod]
+		if len(pkg.types) > 0 {
+			buffer := &bytes.Buffer{}
+			pkg.genCRDHeader(buffer)
+			for _, t := range pkg.types {
+				pkg.genType(buffer, t)
+			}
+			pkg.genTypeRegistrations(buffer, pkg.types)
+			buffers[mod] = buffer
+		}
+	}
+
+	return buffers, nil
+}
+
+func (pkg *pkgContext) genCRDHeader(w io.Writer) {
+	fmt.Fprintf(w, "// *** This file was generated by crd2pulumi. ***\n\n")
```

Comment:
I realize this file is specific to crd2pulumi given its name, but since `GenCRDTypes` already accepts a `tool` arg, might as well pass that along to `genCRDHeader` and use it when emitting the header comment:

```
fmt.Fprintf(w, "// *** This file was generated by %s. ***\n\n", tool)
```

Same feedback applies for nodejs.